### PR TITLE
[v1.x] Rearrange ONNX tests in Nightly CI

### PR DESCRIPTION
**Pull request by waytrue17**
## Description ##
Rearrange ONNX tests in Nightly CI to reduce test time and fix timeout issue: https://jenkins.mxnet-ci.amazon-ml.com/blue/organizations/jenkins/NightlyTestsForBinaries/detail/v1.x/14/pipeline/103. Added `pytest.ini` to document pytest markers

### Essentials ###
- [ ] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Review by josephevans — APPROVED** [review]
LGTM, thanks!

File: ci/docker/runtime_functions.sh [hunk 1]
```diff
@@ -1263,17 +1263,17 @@ integrationtest_ubuntu_cpu_onnx() {
...
+    pytest $COV_ARG --verbose tests/python-pytest/onnx/test_onnxruntime_cv.py::test_obj_class_model_inference_onnxruntime[resnet50_v2]
+    pytest $COV_ARG --verbose tests/python-pytest/onnx/test_onnxruntime_cv.py::test_obj_class_model_inference_onnxruntime[vgg19_bn]
+    pytest $COV_ARG --verbose tests/python-pytest/onnx/test_onnxruntime_cv.py::test_obj_detection_model_inference_onnxruntime[center_net_resnet101_v1b_voc]
+    pytest $COV_ARG --verbose tests/python-pytest/onnx/test_onnxruntime_cv.py::test_img_segmentation_model_inference_onnxruntime[deeplab_resnet50_citys]
+    pytest $COV_ARG --verbose tests/python-pytest/onnx/test_onnxruntime_cv.py::test_pose_estimation_model_inference_onnxruntime[mobile_pose_mobilenet1.0]
+    pytest $COV_ARG --verbose tests/python-pytest/onnx/test_onnxruntime_cv.py::test_action_recognition_model_inference_onnxruntime[inceptionv3_kinetics400]
+    pytest $COV_ARG --verbose tests/python-pytest/onnx/test_onnxruntime_nlp.py::test_dynamic_shape_bert_inference_onnxruntime
+    pytest $COV_ARG --verbose tests/python-pytest/onnx/test_onnxruntime_cv.py::test_dynamic_shape_cv_inference_onnxruntime
```
Comment (szha) [inline ci/docker/runtime_functions.sh:1276]: Why do you need to explicitly list the tests? Alternatively you can choose to mark the tests in the test files and then use the test selection functionality of pytest.

Comment (waytrue17) [reply ci/docker/runtime_functions.sh]: Applied mark for pytest group. Please review @szha

**Comment by szha** [comment]
I think we can use pytest.ini for documenting the tags. Here's the one we are using on the master branch but I didn't find it in 1.x. https://github.com/apache/incubator-mxnet/blob/master/pytest.ini

**Comment by waytrue17** [comment]
> I think we can use pytest.ini for documenting the tags. Here's the one we are using on the master branch but I didn't find it in 1.x. https://github.com/apache/incubator-mxnet/blob/master/pytest.ini

Added `pytest.ini` in v1.x

File: pytest.ini [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+# under the License.
+
+[pytest]
+markers =
+    seed: set the python, numpy and mxnet random seeds to a specified value for test reproducibility
+    serial: mark a test that requires more resources to run that are thus only suitable for serial run.
+    remote_required: mark a test that requires internet access.
+    gpu: mark a test that requires GPU.
+    integration: mark an integration test
+    onnx_coverage: ONNX coverage test
```
Comment (szha) [inline pytest.ini:25]: how is it different from a regular unit test?

File: pytest.ini [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    garbage_expected: this test leaks ndarray references. The tested functionality is broken or there is a Python bug.
+    integrationtest_onnx: mark a test for onnx integration test in CI
```
Comment (szha) [inline pytest.ini:27]: how is it different from regular integration test?

Comment (waytrue17) [reply pytest.ini:27]: Due to the heavy computation load of onnx tests, we only run the full onnx tests in nightly CI, and run a subset of it in the PR integration tests. This marker is used to mark the subset of onnx tests. I can change the tag to something like `onnx_pr_test` if it's more clear.  wdyt?

File: pytest.ini (hunk 2 above)
Comment (waytrue17) [reply pytest.ini:25]: This tag was introduced in master from: https://github.com/apache/incubator-mxnet/pull/18025. I copied it from master, not sure what it is for.

**Comment by waytrue17** [comment]
@mxnet-bot run ci [website]

File: pytest.ini (hunk 2 above)
Comment (szha) [reply pytest.ini:25]: tag is just a name, either for selection or reporting. let's only keep the necessary ones.

File: pytest.ini (hunk 3 above)
Comment (szha) [reply pytest.ini:27]: since you are running the ONNX tests one file at a time, I think using the integration mark should suffice.

Comment (waytrue17) [reply pytest.ini]: Makes sense. Switched to `integration` tag

**Review by Zha0q1 — APPROVED** [review]

**Pull request merged**